nixos/modules/services/mail/maddy.nix

```diff
@@ -0,0 +1,247 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  name = "maddy";
+  cfg = config.services.maddy;
+  defaultConfig = ''
```

Comment:
This is a large block of configuration. Is this your preference, recommendation from upstream, etc...? Can you please leave a comment explaining where/why/how this comes from?